drivers/sensor/wsen_itds/itds.c

```diff
@@ -385,17 +385,23 @@ static const struct sensor_driver_api itds_api = {
 	.channel_get = itds_channel_get,
 };
 
+#ifdef CONFIG_ITDS_TRIGGER
+#define WSEN_ITDS_CFG_IRQ(inst)						\
+	.int_gpio = GPIO_DT_SPEC_INST_GET(inst, int_gpios),
+#else
+#define WSEN_ITDS_CFG_IRQ(inst)
+#endif
+
 #define WSEN_ITDS_INIT(idx)						\
 									\
 static struct itds_device_data itds_data_##idx;				\
 									\
 static const struct itds_device_config itds_config_##idx = {		\
 	.i2c = I2C_DT_SPEC_INST_GET(idx),				\
-	.gpio_port = DT_INST_GPIO_LABEL(idx, int_gpios),		\
-	.int_pin = DT_INST_GPIO_PIN(idx, int_gpios),			\
-	.int_flags = DT_INST_GPIO_FLAGS(idx, int_gpios),		\
 	.def_odr = DT_INST_ENUM_IDX(idx, odr),				\
 	.def_op_mode = DT_INST_ENUM_IDX(idx, op_mode),			\
+	COND_CODE_1(DT_INST_NODE_HAS_PROP(idx, int_gpios),		\
+		(WSEN_ITDS_CFG_IRQ(idx)), ())				\
```

Comment:
suggestion:

```c
IF_ENABLED(CONFIG_ITDS_TRIGGER, (.int_gpio = GPIO_DT_SPEC_INST_GET(inst, int_gpios),))
```

If int_gpio is optional even with `CONFIG_ITDS_TRIGGER=y` use `GPIO_DT_SPEC_INST_GET_OR` and adjust init code accordingly.

Reply:
Update the code here, I went with `GPIO_DT_SPEC_INST_GET_OR` since we could have multiple instances of the device and some support / have `int_gpio` in dts I didn't consider it an error if the prop was missing.